src/server.js

```diff
@@ -0,0 +1,249 @@
+		const meshConfigPath = `${currentWorkingDirectory}/mesh-artifact/${meshId}/.meshrc.json`;
+
+		const meshConfig = require(meshConfigPath);
+		const { responseConfig } = meshConfig;
+		const { CORS } = responseConfig;
+
+		return CORS;
+	} catch (e) {
+		return {};
+	}
+};
+
+const getYogaServer = async () => {
+	if (yogaServer) {
+		return yogaServer;
+	} else {
+		const currentWorkingDirectory = process.cwd();
+		const meshArtifactsPath = `${currentWorkingDirectory}/mesh-artifact/${meshId}`;
+
+		const meshArtifacts = require(meshArtifactsPath);
+		const { getBuiltMesh } = meshArtifacts;
+
+		const tenantMesh = await getBuiltMesh();
+		const corsOptions = getCORSOptions();
+
+		logger.info('Creating graphQL server');
+
+		meshConfig = readMeshConfig(meshId);
+
+		if (isTI === 'true') {
+			// TI customers get access to fetcher and sessionCache
+			//create context config
+			const { fetchConfig, cacheConfig } = processContextConfig(meshId, meshConfig);
+			const contextCache = new LRU(cacheConfig);
+
+			const allowedDomainsMap = fetchConfig.allowedDomains?.reduce((acc, allowedDomain) => {
+				acc[allowedDomain] = {};
+				logger.info(`acc: ${acc}`);
+				return acc;
+			}, {});
+
+			yogaServer = createYoga({
+				cors: corsOptions,
+				plugins: tenantMesh.plugins,
+				graphqlEndpoint: '/graphql',
+				graphiql: false,
+				maskedErrors: false,
+				context: initialContext => {
+					return {
+						...initialContext,
+						sessionCache: contextCache,
+						log: message => logger.info(`${meshId} - ${message}`),
+						fetcher: async (url, options) => {
+							const { protocol, host } = URL.parse(url);
+							if (protocol !== 'https:') {
+								throw new Error(`${url} is not a valid https url`);
+							}
+							const basePath = protocol + '//' + host;
+							logger.info(`Host: ${host}`);
+							logger.info(`Absolute base: ${basePath}`);
+							logger.info(`allowedDomainsMap: ${JSON.stringify(allowedDomainsMap)}`);
+
+							if (basePath !== null && allowedDomainsMap !== null) {
+								if (!(basePath in allowedDomainsMap)) {
+									logger.info(
+										`host: ${host} and allowedDomainsMap: ${allowedDomainsMap} and stringified allowedDomain: ${JSON.stringify(
+											allowedDomainsMap,
+										)}`,
+									);
+									logger.error(`${url} is not allowed to be accessed`);
+									throw new Error(`${url} is not allowed to be accessed`);
+								} else {
+									logger.info(
+										`Fetching invokeRemoteFetch ${url} with options ${JSON.stringify(options)}`,
+									);
+									const response = await invokeRemoteFetch(url, options);
+									const body = await response.text();
+									logger.info(`Fetched ${url}. Response body: ${body}`);
+									return { response, body };
+								}
+							}
+						},
+					};
+				},
+			});
+		} else {
+			yogaServer = createYoga({
+				plugins: tenantMesh.plugins,
+				graphqlEndpoint: `/graphql`,
+				graphiql: false,
```

Comment:
Might the built-in GraphiQL client be useful for local development?

Reply:
Yup, good call, lets enable it for local development.

```suggestion
				graphiql: true,
```